File: app/src/main/java/ga/chrom_web/player/multiplayer/data/ChatMessage.java
package ga.chrom_web.player.multiplayer.data;


import android.graphics.Color;

public class ChatMessage extends ChatItem {
    private String text;
    private String color;

    public ChatMessage() {
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public int getColor() {
        return Color.parseColor(color);
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getFullMessage() {
        return nick + ": " + text;
    }
}

File: app/src/main/java/ga/chrom_web/player/multiplayer/data/ChatMessage.kt
package ga.chrom_web.player.multiplayer.data


import android.graphics.Color
import java.io.Serializable

class ChatMessage : ChatItem(), Serializable {

    companion object {
        private const val DEFAULT_COLOR = Color.BLACK
    }

    private var text: String? = null
    private var color: String? = null

    fun getColor(): Int {
        color?.let {
            return Color.parseColor(it)
        }
        return DEFAULT_COLOR;
    }

    fun getFullMessage():String = nick + ": " + text

    fun getText(): String? = text

    override fun toString(): String = "ChatMessage(text=$text, color=$color)"
}
